GVFS/GVFS/CommandLine/StatisticsVerb.cs

```diff
@@ -0,0 +1,298 @@
+﻿using CommandLine;
+using GVFS.Common;
+using GVFS.Common.Database;
+using GVFS.Common.FileSystem;
+using GVFS.Common.Git;
+using GVFS.Common.NamedPipes;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace GVFS.CommandLine
+{
+    [Verb(StatisticsVerb.StatisticsVerbName, HelpText = "Get statistics for the health state of the repository")]
+    public class StatisticsVerb : GVFSVerb.ForExistingEnlistment
+    {
+        private const string StatisticsVerbName = "statistics";
+
+        private const int DirectoryDisplayCount = 5;
+
+        [Option(
+            'c',
+            "crawl",
+            Required = false,
+            HelpText = "Hydrate all files (for testing purposes)")]
+        public bool ToCrawl { get; set; }
+
+        protected override string VerbName
+        {
+            get { return StatisticsVerbName; }
+        }
+
+        protected override void Execute(GVFSEnlistment enlistment)
+        {
+            // Debugging entry point to force the creation of placeholders
+            if (this.ToCrawl)
+            {
+                this.Crawl();
+            }
+
+            this.GetPlaceholdersFromDatabase(enlistment, out List<IPlaceholderData> filePlaceholders, out List<IPlaceholderData> folderPlaceholders);
+            string[] modifiedPathsList = this.GetModifiedPathsFromPipe(enlistment);
+
+            /* Failed code for creating a local instance of the modified paths database
+            using (JsonTracer tracer = new JsonTracer(GVFSConstants.GVFSEtwProviderName, "Statistics"))
+            {
+                ModifiedPathsDatabase.TryLoadOrCreate(
+                    tracer,
+                    Path.Combine(enlistment.DotGVFSRoot, GVFSConstants.DotGVFS.Databases.ModifiedPaths),
+                    new PhysicalFileSystem(),
+                    out ModifiedPathsDatabase modifiedPathsDatabase,
+                    out string err);
+            }
+            */
+
+            int placeholderCount = filePlaceholders.Count + folderPlaceholders.Count;

```

Comment:
Can we put the business logic for the computations into its own class? That way the verb can be responsible for the reading / parsing input parameters, but then leave the details of the business logic to another component. This will also allow us to test the business logic class directly.

Reply:
Where would this component live?

Reply:
In the `GVFS.Common` project. At this point, I would leave it in the top level directory (unless / until there will be a family of related components)